File: PSET 6/caesar.py
import cs50
import sys

def main():
    if len(sys.argv) != 2:
        print("Usage: caesar.py k")
        exit(1)

    if len(sys.argv) == 2:
        key = int(sys.argv[1])
        key = key%26
        print("plaintext:  ", end="")
        plain = cs50.get_string() # plain asks for "p"lain text
        print("ciphertext: ", end="")

    if plain != None:
        n = len(plain)
        for char in plain: #sweep the length of plain text
            if not char.isalpha(): # if it is not a letter, re-print as is
                print(char, end="")
            if char.isupper():
                print(chr((ord(char)-65+key)%26+65), end="")
                # https://stackoverflow.com/questions/227459/ascii-value-of-a-character-in-python
                # ord() to convert from char to int
                # chr() to convert int back to char
            if char.islower():
                print(chr((ord(char)-95+key)%26+65), end="")
        print("")
    exit(0)

if __name__ == "__main__":
    main()

File: PSET 2/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <stdlib.h>

int main(int argc, string argv[])
{
    string key = argv[1];
    if (argc !=2)
    {
        printf("Usage: ./caesar k\n");
        return 1;
    }
    if (argc == 2)
    {
        int k = atoi(key);
        k = k % 26;
        printf("plaintext:  ");
        string p = get_string(); // p asks for "p"lain text
        printf("ciphertext: ")  ;
        if (p !=NULL)
        {
            int n = strlen(p);
            for (int i = 0; i < n; i++) // sweep the length of P
            {
                if (!isalpha(p[i])) // if it is not a letter, re-print as is
                {
                    printf("%c", p[i]);
                }
                if (isupper(p[i]))
                {
                    p[i]=((p[i]-65)+k)%26+65;
                    printf("%c", (char)p[i]);
                }
                if (islower(p[i]))
                {
                    p[i]=((p[i]-97)+k)%26+97;
                    printf("%c", (char)p[i]);
                }
            }
            printf("\n");
        }
    return 0;
    }
}